.github/workflows/formatCheck.yml

```diff
@@ -0,0 +1,18 @@
+name: Format check
```

Comment:
Shall we call this `build.yml` and add both (1) checking format  and (2) build for compilation errors

Later we can also add unit tests.